store/badger.go

```diff
@@ -132,6 +154,39 @@ func (b *badgerKV) CommitWriteBatch(batch WriteBatch) error {
 	return wb.batch.Flush()
 }
 
+func (b *badgerKV) gc(interval time.Duration) {
+	b.closeWg.Add(1)
+
+	go func() {
+		defer b.closeWg.Done()
+
+		ticker := time.NewTicker(interval)
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-b.closeCh:
+				return
+			case <-ticker.C:
+				for {
+					// Check for close signal, in case it's closed during GC loop.
+					select {
+					case <-b.closeCh:
+						return
+					default:
+					}
+
+					// When there's no more GC, an error will be returned.
+					err := b.db.RunValueLogGC(0.5)
```

Comment:
I would suggest you reduce the `discardRatio`. A ratio of `0.5` means that `Delete the file only if 50% of the file contains invalid data`. It would be difficult to get a discard ration of 0.5 unless you have equal number of read/write happening.

Reply:
Changed to 0.1.